### PR TITLE
[1/3] refactor: port some misc changes from upstream

### PR DESCRIPTION
**Pull request by div72**

### CODE REVIEW

**Comment by barton2526** [comment]
Can you link the relevant upstream PRs/commits?

**Comment by div72** [comment]
I can try to, but not all of these changes belong to a full PR.

File: src/base58.cpp [hunk 1]
```diff
@@ -161,7 +161,7 @@ std::string EncodeBase58Check(const std::vector<unsigned char>& vchIn)
     }
     // re-calculate the checksum, ensure it matches the included 4-byte checksum
     // Can't use spans here until we teach hash.h about them.
```
Comment (jamescowens) [inline src/base58.cpp]: Let's fix my earlier comment here since it is not relevant anymore.

File: src/serialize.h [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) 2009-2010 Satoshi Nakamoto
 // Copyright (c) 2009-2018 The Bitcoin Core developers
 // Distributed under the MIT software license, see the accompanying
-// file COPYING or https://opensource.org/licenses/mit-license.php.
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
```
Comment (jamescowens) [inline src/serialize.h]: Upstream is wrong here I think. THEY need to fix this. It should always be https I think if https works.

File: src/uint256.cpp [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) 2009-2010 Satoshi Nakamoto
-// Copyright (c) 2009-2018 The Bitcoin Core developers
+// Copyright (c) 2009-2020 The Bitcoin Core developers
 // Distributed under the MIT software license, see the accompanying
-// file COPYING or https://opensource.org/licenses/mit-license.php.
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
```
Comment (jamescowens) [inline src/uint256.cpp]: https

**Review by jamescowens — COMMENTED** [review]
You should verify in all places that the copy of header files from upstream is not reverting https to http.

**Review by jamescowens — APPROVED** [review]
Nice work. These are a pain I know... :)

**Comment by jamescowens** [comment]
This is an ideal time to merge this stuff while there is no big outstanding PR to conflict...

File: src/hash.h [hunk 4]
```diff
@@ -1,21 +1,21 @@
 // Copyright (c) 2009-2010 Satoshi Nakamoto
-// Copyright (c) 2009-2017 The Bitcoin Core developers
-// Copyright (c) 2019-2021 The Gridcoin Core Developers
+// Copyright (c) 2009-2020 The Bitcoin Core developers
```
Comment (barton2526) [inline src/hash.h:2]: nit, we lost Gridcoin developers here

Comment (div72) [reply src/hash.h:2]: We don't have any original code in this file other than the http -> https change for the copyright header.

Comment (jamescowens) [reply src/hash.h:2]: I agree.

**Pull request merged**